src/resultset/resultset.c

```diff
@@ -93,8 +93,8 @@ void ResultSet_MapProjection(ResultSet *set, const Record r) {
 static void _ResultSet_ReplyWithPreamble(ResultSet *set, const Record r) {
 	ASSERT(set->recordCount == 0);
 
-	// Prepare a response containing a header, records, and statistics
-	RedisModule_ReplyWithArray(set->ctx, 3);
+	// Prepare a response containing a header, records, statistics, and possibly a metadata footer.
+	RedisModule_ReplyWithArray(set->ctx, REDISMODULE_POSTPONED_ARRAY_LEN);
```

Comment:
Let's avoid the postponed array length and reply with array of length 4, wort case we will emit an empty map in the footer

Reply:
I'd prefer to leave this as-is for now, since that would expose the metadata footer to all users (not just those using the `version` flag) as well as breaking run-time error logic.

Once your work on `avoid-postponed-array` is merged, we can easily refactor this to remove the postponed call.